test/delivery/delivery_test.go

```diff
@@ -68,6 +68,17 @@ var _ = Describe("delivery", Ordered, func() {
 				return attempt.Status == entities.AttemptStatusSuccess
 			}, time.Second*15, time.Second)
 
+			var attemptDetail *entities.AttemptDetail
+			assert.Eventually(GinkgoT(), func() bool {
+				val, err := db.AttemptDetails.Get(context.TODO(), attempt.ID)
+				if err != nil || val == nil {
+					return false
+				}
+				attemptDetail = val
+				return true
+			}, time.Second*5, time.Second)
+			attempt.Extend(attemptDetail)
```

Comment:
```suggestion
```

Reply:
Please modify the following tests to do

1. assert attempt.request.headers, attempt.request.body, attempt.response.headers, attempt.response.body are nil as these should be transfer to AttemptDetail
2. assert fields in that attemptDetails are expected.

Reply:
updated